Engine/DataFeeds/DownloaderDataProvider.cs

```diff
@@ -104,19 +121,38 @@ public override Stream Fetch(string key)
                         // since hourly & daily are a single file we fetch the whole file
                         try
                         {
-                            startTimeUtc = symbol.ID.Date;
+                            // we don't really know when Futures, FutureOptions, Cryptos, etc, start date so let's give it a good guess
+                            if (symbol.SecurityType == SecurityType.Crypto)
+                            {
+                                // bitcoin start
+                                startTimeUtc = new DateTime(2009, 1, 1);
+                            }
+                            else if (symbol.SecurityType.IsOption() && symbol.Underlying != null && symbol.SecurityType == SecurityType.Option)
+                            {
+                                // the underlying equity start date
+                                startTimeUtc = symbol.Underlying.ID.Date;
```

Comment:
We can have mixed companies in the same file e.g. FB was an insurance company I think and later FB social media (before renaming to META). I think we segment options by year so it would be a bug case if the ticker were reused within a year.